### DISCUSSION TITLE
Base64 issue

### DISCUSSION THREAD

**Discussion by EthanKC** [nextgenhealthcare/connect#5215] [Q&A] [answered]
I have a situation where an ORU message is being sent to me with the OBX.5.1 as a long report string. The receiving side needs it as a PDF encoded in base64. I can encode that string as base64, but it's not a PDF. Any ideas how I can extract that string, turn it into PDF format, encode it as base64, then insert it back into the message? I've been trying things with multiple channels, using a PDF writer to write the OBX.5.1 to a PDF, then picking that up and writing it to a database. At the same time, I write the first part of the message, up to the OBX.4, to a different field in the database, and OBX.6 to the end in a third field. Then I tried using a database reader to pull all three fields and combine them, but I'm running into issues. I've attached what I have so far, but I suspect this isn't the best way to do this. Any ideas? (couldn't upload the .XML, so I attached it as a .txt)
[Base64 Stuff.txt](https://github.com/nextgenhealthcare/connect/files/8771515/Base64.Stuff.txt)

**Comment by pacmano1** [comment, marked as answer]
see https://forums.mirthproject.io/forum/mirth-connect/support/181200-transform-base-64-image-to-base64-pdf

The formats and messages type  are different, but the concepts are the same as it applies to creating the PDF in the channel for downstream use.

**Reply by EthanKC** [reply]
Thanks, that's super helpful! I created a document writer that is writing the OBX.5.1 to a PDF. The only thing I can't figure out is how to reference that attachment in another connector. Everything I'm finding says to use the attachment ID from the response, but I'm not seeing/can't figure out how to actually do that.

**Reply by pacmano1** [reply]
https://forums.mirthproject.io/articles/175046-using-attachments, that is an awesome article on using attachments.

**Reply by EthanKC** [reply]
Yeah, I'd found that before, but somehow this time it actually clicked. I just got it to work! Thanks again for the help!
[ORU from Src.txt](https://github.com/nextgenhealthcare/connect/files/8772425/ORU.from.Src.txt)
